subprojects/asio.wrap

```diff
@@ -1,8 +1,8 @@
 [wrap-file]
-directory = asio-1.24.0
-source_url = https://sourceforge.net/projects/asio/files/asio/1.24.0%20%28Stable%29/asio-1.24.0.tar.gz/download
-source_filename = asio-1.24.0.tar.gz
-source_hash = 2f23ef6eada06ecc1472af5df6365ed4f15452ccd07dc0a6851fa20d571dba94 
+directory = asio-asio-1-24-0
+source_url = https://github.com/chriskohlhoff/asio/archive/refs/tags/asio-1-24-0.tar.gz
```

Comment:
@neheb, @eli-schwartz : what would be a recommended solution to handle this wrapdb sanity check error? As you can see asio upstream project uses somewhat specific versioning scheme diverging between tag names (1-24-0) and release versions (1.24.0).
```
AssertionError: False is not true : Version 1.24.0 not found in https://github.com/chriskohlhoff/asio/archive/refs/tags/asio-1-24-0.tar.gz
```

Reply:
Answering for myself.... There are seems to be couple other projects with similar problem, so, that's handled as exception right in the sanity check script:
https://github.com/mesonbuild/wrapdb/blob/b160efaee3d15614ab3cb0ff26086e14f62c1c5c/tools/sanity_checks.py#L305-L309

I'll add similar exception for asio.